radlab-ui/webapp/src/components/forms/StepCreator.tsx

```diff
@@ -59,24 +60,35 @@ const StepCreator: React.FC<StepCreator> = ({ variableList, idx }) => {
     return error
   }
   const renderControls = (variable: IUIVariable) => {
-    switch (variable.type) {
-      case "string":
-        return <StringField variable={variable} validate={validate(variable)} />
-      case "bool":
-        return <BooleanField variable={variable} />
-      case "number":
-        return <NumberField variable={variable} validate={validate(variable)} />
-      case "set(string)":
-      case "set(number)":
-        return <SetField variable={variable} />
-      case "list(string)":
-      case "list(number)":
-        return <ListField variable={variable} />
-      case "map":
-      case "map(string)":
-        return <MapField variable={variable} validate={validate(variable)} />
-      default:
-        return <StringField variable={variable} validate={validateRequired} />
+    let checkMapObject = variable.type.slice(0, 10)
+    if (checkMapObject === "map(object") {
+      return (
+        <MapNestedField variable={variable} validate={validate(variable)} />
+      )
+    } else {
```

Comment:
No need for the nested `else` since you're `return`ing in the `if` above.

Reply:
Changes will be made in a separate PR based on your feedback.